pkg/vdr/key/creator.go

```diff
@@ -88,10 +97,82 @@ func (v *VDR) Create(keyManager kms.KeyManager, didDoc *did.Doc,
 	return &did.DocResolution{DIDDocument: createDoc(publicKey, keyAgr, didKey)}, nil
 }
 
+func getKeyTypeAndVerMethodType(kt kms.KeyType) (kms.KeyType, string) {
+	var (
+		keyType kms.KeyType
+		vmType  string
+	)
+
+	switch kt {
+	case kms.ED25519Type:
+		keyType = kms.ED25519Type
+		vmType = ed25519VerificationKey2018
+	case kms.ECDSAP256TypeDER:
+		keyType = kms.ECDSAP256TypeDER
+		vmType = jsonWebKey2020
+	case kms.ECDSAP256TypeIEEEP1363:
+		keyType = kms.ECDSAP256TypeIEEEP1363
+		vmType = jsonWebKey2020
+	case kms.ECDSAP384TypeDER:
+		keyType = kms.ECDSAP384TypeDER
+		vmType = jsonWebKey2020
+	case kms.ECDSAP384TypeIEEEP1363:
+		keyType = kms.ECDSAP384TypeIEEEP1363
+		vmType = jsonWebKey2020
+	case kms.ECDSAP521TypeDER:
+		keyType = kms.ECDSAP521TypeDER
+		vmType = jsonWebKey2020
+	case kms.ECDSAP521TypeIEEEP1363:
+		keyType = kms.ECDSAP521TypeIEEEP1363
+		vmType = jsonWebKey2020
+	case kms.BLS12381G2Type:
+		keyType = kms.BLS12381G2Type
+		vmType = bls12381G2Key2020
+	default:
+		keyType = kms.ED25519Type
+		vmType = ed25519VerificationKey2018
+	}
+
+	return keyType, vmType
+}
+
+func getKeyCode(keyType kms.KeyType, verificationMethod *did.VerificationMethod) (uint64, error) {
```

Comment:
Note: the reverse is also needed... translate a verificationMethod into a `kms.KeyType`: https://github.com/llorllale/edge-service/blob/44ebaad708cd1bc2094b1d70ac72813cfefd8f1c/pkg/restapi/csh/operation/zcapld/zcapld.go#L205-L237

Reply:
fyi, JWS signing works with keys encoded in IEEE_P1363 format, verifying with ASN.1 DER key format don't work with JWS signatures, this is why we recommend using DER ECDSA key types.

Reply:
also my change allows for creating ECDSA  keys in both formats, the logic from your link only supports one format (DER).

this is why doing a switch over the `keyType` is needed

Reply:
Yes, the link was provided to show that this kind of conversion is needed. Improvements are welcome.

Reply:
the logic of getKeyCode() is now enhanced:
* if keyType option was passed, the key code will be determined by keyType
* else the VerificationMethod.Value containing the raw key is used to determine the key code based on the raw key's length (for JSONWebKey2020 keys assuming they're only ECDSA keys for now, this can be enhanced later to support other keys as JSONWebKey2020, the challenge is we can't determine the key type from the raw key value alone so for now we assume JSONWebKey2020 are ECDSA keys).

The reverse is available via `fingerprint.PubKeyFromFingerPrint(fingerprint string) ([]byte, uint64, error) ` [here](https://github.com/hyperledger/aries-framework-go/pull/2594/files#diff-db294652c6ab1f742e49369b6ad733d1f4a3312dc0072e08bd8aed2ae43a92f8R71)

Reply:
> else the VerificationMethod.Value containing the raw key is used to determine the key code based on the raw key's length (for JSONWebKey2020 keys assuming they're only ECDSA keys for now, this can be enhanced later to support other keys as JSONWebKey2020, the challenge is we can't determine the key type from the raw key value alone so for now we assume JSONWebKey2020 are ECDSA keys).

JsonWebKey2020 is just a JWK and has the `crv` attribute... [example](https://w3c-ccg.github.io/lds-jws2020/#example-1). To determine whether the use is signature or key agreement, you would check the verification method relationship (basically, all verification method relationships are for signing except `keyAgreement`).

Reply:
See [this table](https://w3c-ccg.github.io/lds-jws2020/#jose-conformance).

Reply:
Agreed, the logic needs to be enhanced by fetching the JWK of that verification method (`verificationMethod.JSONWebKey()`), with this we can detect the raw key type and multicodec key code. 

This can be done in a followup change.